src/views/FarmAuction/hooks/useReclaimAuctionBid.ts

```diff
@@ -0,0 +1,154 @@
+import { useEffect, useReducer } from 'react'
+import { useWeb3React } from '@web3-react/core'
+import BigNumber from 'bignumber.js'
+import { BidderAuction } from 'config/constants/types'
+import { useFarmAuctionContract } from 'hooks/useContract'
+import { RECLAIM_AUCTIONS_TO_FETCH } from 'config'
+import { processBidderAuctions, sortAuctionBidders } from '../helpers'
+
+interface ReclaimableAuction {
+  id: number
+  amount: BigNumber
+  position: number
+}
+
+interface ReclaimReducerState {
+  auctions: BidderAuction[]
+  currentCursor: number
+  nextCursor: number
+  nextAuctionToCheck: number // nextAuctionToCheck is array index in auctions
+  auctionToReclaim: ReclaimableAuction | null
+  loading: boolean
+}
+
+const initialState: ReclaimReducerState = {
+  auctions: [],
+  currentCursor: 0,
+  nextCursor: 0,
+  nextAuctionToCheck: 0,
+  auctionToReclaim: null,
+  loading: false,
+}
+
+const reclaimReducer = (state: ReclaimReducerState, action: { type: string; payload?: any }): ReclaimReducerState => {
+  switch (action.type) {
+    case 'setAuctions':
+      return {
+        auctions: action.payload.auctions,
+        currentCursor: state.currentCursor,
+        nextCursor: action.payload.nextCursor,
+        nextAuctionToCheck: 0,
+        auctionToReclaim: null,
+        loading: false,
+      }
+    case 'setAuctionToReclaim':
+      return {
+        ...state,
+        auctionToReclaim: action.payload.auctionToReclaim,
+        loading: false,
+      }
+    case 'checkNextAuction': {
+      const nextAuctionToCheck = state.nextAuctionToCheck + 1
+      if (nextAuctionToCheck === state.auctions.length) {
+        // Checked all auctions in the batch
+        return {
+          ...state,
+          auctions: [],
+          currentCursor: state.nextCursor,
+          auctionToReclaim: null,
+          loading: false,
+        }
+      }
+      return {
+        ...state,
+        nextAuctionToCheck,
+        auctionToReclaim: null,
+        loading: false,
+      }
+    }
+    case 'setLoading':
+      return { ...state, loading: action.payload.loading }
+    case 'reset':
+      return initialState
+    default:
+      return state
+  }
+}
+
+/**
+ * This hook checks if user has participated in previous auctions and has some bids to claim back.
+ */
+const useReclaimAuctionBid = (): [ReclaimableAuction | null, () => void] => {
```

Comment:
This hook is used to determine if user has any bids to reclaim. If reclaimable bids found - show ReclaimBidCard with one of the reclaimable auction. Once user successfully reclaims auction - show ReclaimBidCard for another one. If there is nothing to reclaim card is not shown.